src/main/java/it/pagopa/interop/probing/telemetry/writer/annotations/validator/TelemetryValidator.java

```diff
@@ -0,0 +1,21 @@
+package it.pagopa.interop.probing.telemetry.writer.annotations.validator;
+
+import java.util.Objects;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import it.pagopa.interop.probing.telemetry.writer.annotations.ValidateTelemetry;
+import it.pagopa.interop.probing.telemetry.writer.dto.TelemetryDto;
+import it.pagopa.interop.probing.telemetry.writer.util.EserviceStatus;
+
+public class TelemetryValidator implements ConstraintValidator<ValidateTelemetry, TelemetryDto> {
+
+  @Override
+  public boolean isValid(TelemetryDto dto, ConstraintValidatorContext context) {
+    return Objects.isNull(dto.getStatus())
```

Comment:
yes but the notNull check is already be made in the @NotNull annotation on status. This annotation does a different check, so in case of a null it does result valid , but the @NotNull error is raised.

Reply:
I'm not sure to follow, it's ok that the nullability is already checked, but given the scope of this function, if the status is null the record is not valid, right?

Reply:
Yes the record should not be valid, but this is not the purpose of this validation. In case of status null, then it would be considered not valid and the error "A telemetry with an OK status must have a response time and no ko reason, while one with a KO status must have a ko reason and no response time" would be thrown followed by not null error, which i think is incorrect. So i divided the two behaviours by adding the IsNull check at the start. Maybe i can change the name of the validation to be more specific about the fields it refers.